.github/workflows/code-check.yml

```diff
@@ -0,0 +1,160 @@
+      matrix:
+        feature: [ libp2p ]
+        os: [ self-hosted, macos-latest ] # drop windows for now as risc0 does not support it
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+      - name: Install protoc compiler
+        uses: arduino/setup-protoc@v3
+        with:
+          repo-token: ${{ secrets.GITHUB_TOKEN }}
+      - name: Setup Cargo cache
+        uses: ./.github/actions/setup-cargo-cache
+        with:
+          # We use the same cache key for both OSes
+          key: ${{ github.ref }}->${{ github.workflow }}->${{ github.job }}
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+      - name: Install cargo-binstall
+        uses: cargo-bins/cargo-binstall@main
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - name: Install risc0
+        run: cargo binstall -y cargo-risczero && cargo risczero install
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      # TODO: Investigate how to get rid of this seemingly necessary build step before running the tests
+      - name: Cargo build
+        uses: actions-rs/cargo@v1
+        with:
+          command: build
+          args: --all --all-targets --no-default-features --features ${{ matrix.feature }}
+      # We store the result of `cargo build` in the cache, that is then re-used in the next run, and can be picked up by the next `cargo test` step.
+      - name: Update Cargo cache
+        if: success() || failure()
+        uses: ./.github/actions/update-cargo-cache
+        with:
+          key: ${{ github.ref }}->${{ github.workflow }}->${{ github.job }}          
+      - name: Cargo test
+        uses: actions-rs/cargo@v1
+        env:
+          RISC0_DEV_MODE: true
+          CONSENSUS_SLOT_TIME: 5
+        with:
+          command: test
+          args: --all --no-default-features --features ${{ matrix.feature }}
+      - name: Cargo bench (no-run)
+        uses: actions-rs/cargo@v1
+        env:
+          RISC0_DEV_MODE: true
+          CONSENSUS_SLOT_TIME: 5
+        with:
+          command: bench
+          args: --no-run
+      - name: Upload integration tests results
+        uses: actions/upload-artifact@v4
+        if: failure()
+        with:
+          name: integration-test-artifacts
+          path: tests/.tmp*
+  
+  build-docker:
+    name: Trigger Docker build
+    runs-on: ubuntu-latest
+    needs:
+      - formatting
+      - cargo-deny
+      - lints
+      - tests
+    env:
+      DOCKER_BUILD_WORKFLOW_FILE: build-docker.yml # Change this in case of file renamings
+    steps:
+      - name: Trigger "Docker build" workflow
+        uses: actions/github-script@v7
+        with:
+          script: |
+            github.rest.actions.createWorkflowDispatch({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              workflow_id: '${{ env.DOCKER_BUILD_WORKFLOW_FILE }}',
+              ref: '${{ github.ref }}',
```

Comment:
From the failed action output it seems that `github.ref` should be changed to the commit hash here.

Reply:
In the codebase I was working on before (https://github.com/KILTprotocol/kilt-node/blob/67c3d65a8adb6b0ac9578492d882c1f9f117fcdd/.github/workflows/check-code.yml#L381), I did exactly the same for a job that is started on both PRs and push events. I believe the error is due to the fact that the new `workflow_dispatch` configuration for the Docker build workflow is not part of the default branch just yet, otherwise I would not be able to explain why this would not work while in the other repo it does.